### PR TITLE
network-services: Added aws_v4_authentication field to EdgeCacheOrigin

### PR DESCRIPTION
**Pull request by carlyeks**
Adds `aws_v4_authentication` field to the `EdgeCacheOrigin` to support [private S3-compatible origins](https://cloud.google.com/media-cdn/docs/connect-to-s3-compatible-buckets).
Resolves hashicorp/terraform-provider-google/issues/12862. 

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
network_services: Added `aws_v4_authentication ` field to `google_network_services_edge_cache_origin ` to support S3-compatible Origins
```

### CODE REVIEW

File: mmv1/products/networkservices/terraform.yaml [hunk 1]
```diff
@@ -45,6 +45,14 @@ overrides: !ruby/object:Overrides::ResourceOverrides
           resource_name_2: "my-fallback"
         ignore_read_extra:
          - "timeout"
+      - !ruby/object:Provider::Terraform::Examples
+        name: "network_services_edge_cache_origin_v4auth"
+        primary_resource_id: "default"
+        vars:
+          resource_name: "my-origin"
+          access_key_id: "ACCESSKEYID"
```
Comment (rileykarson) [inline mmv1/products/networkservices/terraform.yaml:53]: nit: We likely want to bake this value into the config, `vars` is mostly for resource names, which need to be unique.

**Review by rileykarson — COMMENTED** [review]
One small nitpick, otherwise LGTM.

File: mmv1/products/networkservices/terraform.yaml (hunk 1 above)
Comment (carlyeks) [reply mmv1/products/networkservices/terraform.yaml]: Thanks for pointing that out - removed the `access_key_id` var, but added `secret_name` which could cause issues if not unique.

**Comment by carlyeks** [comment]
@rileykarson: gentle ping for a review here.

**Comment by carlyeks** [comment]
@rileykarson Ping for review here.

**Review by rileykarson — APPROVED** [review]
Sorry for the delay! Busy week on my end. LTGM.

**Pull request merged**